ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/DecoderOptions.java

```diff
@@ -21,6 +21,12 @@ public interface DecoderOptions extends SinkOptions, PipelineOptions {
 
   void setProxyIps(String value);
 
+  @Description("Path (local or gs://) to newline delimited text file of regexes matching ip "
+      + "addresses associated with load balancers and proxies that will be ignored for IP lookups.")
+  String getProxySubnetsPath();
```

Comment:
@whd I assume you don't actually want to provide subnets in regex format, could you give an example of the format you want for providing proxy subnets?

Reply:
Ideally I'd be providing the files as-is:

cloudfront: https://d7uri8nf7uskq.cloudfront.net/tools/list-cloudfront-ips

(from https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/LocationsOfEdgeServers.html)

And for GCP:
https://www.gstatic.com/ipranges/cloud.json
https://www.gstatic.com/ipranges/goog.json

From https://support.google.com/a/answer/10026322?hl=en / https://cloud.google.com/vpc/docs/configure-private-google-access

Unfortunately GCP doesn't appear to give CDN-specific allocations, and requires additional parsing to diff user-allocated and service-allocated ranges: https://cloud.google.com/vpc/docs/configure-private-google-access#ip-addr-defaults. I think trusting any GCP-provided IP that isn't allocated to customers would be acceptable but it's probably worth double checking that.

We've used the script they provided before https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-pioneer/tf/modules/study-project/compute_routes in some terraform config, but I would prefer if ops infra is responsible solely for versioning these artifacts as they are provided by upstream providers, and have it be the application's responsibility to parse them.

Reply:
> And for GCP:
> https://www.gstatic.com/ipranges/cloud.json
> https://www.gstatic.com/ipranges/goog.json

combining these in Java is significantly more complicated than it is in Python. I think it would be better to either re-use that script, or to have a copy of it in this repo that is used to prepare the list of non-client google blocks that is passed to the beam jobs.

Reply:
This appears to be published as https://github.com/GoogleCloudPlatform/networking-tools-python/tree/main/tools/cidr#how-to-run. I think it would be fine to use that in ops infra as the canonical way of generating the artifact, but it produces output that isn't really meant to be machine readable, though it looks like if you chop the first 3 lines it probably is.

Otherwise, we need to standardize on a format for the output of such a script, and we're back to https://github.com/mozilla/gcp-ingestion/pull/2271#discussion_r1101906259. I don't have an opinion beyond wanting to maintain the least amount of bespoke code as is reasonable. If we need to process into a format I suppose we could reuse the the format https://d7uri8nf7uskq.cloudfront.net/tools/list-cloudfront-ips uses.

Reply:
> it produces output that isn't really meant to be machine readable, though it looks like if you chop the first 3 lines it probably is

I think that will be good enough for me, i'll write this assuming i need to do the chopping myself.

Reply:
for google the version info is in the header of the computed file, for cloudfront version would you like me to use the filename, a separate cli argument, the file mtime, or some combination of those?